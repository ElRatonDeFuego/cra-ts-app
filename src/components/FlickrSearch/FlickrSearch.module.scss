.flickr-search {
  font-size: calc(10px + 2vmin);

  > button,
  input {
    font-size: calc(10px + 1vmin);
  }
}
