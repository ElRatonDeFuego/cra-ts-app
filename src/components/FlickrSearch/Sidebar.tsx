import {
  faBars,
  faCakeCandles,
  faClose,
} from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { useState } from "react";

export const Sidebar = () => {
  const [isOpen, setIsOpen] = useState(false);

  return (
    <span className="text-emerald-50 opacity-95">
      <div
        className={`fixed top-0 right-0 p-6 pt-20 min-h-screen ease-in-out
                    duration-300 bg-slate-800 flex flex-col items-center
                    ${isOpen ? "translate-x-0" : "translate-x-full"}`}
      >
        <h2>Useless animated sidebar!</h2>
        <FontAwesomeIcon
          className="text-center mt-8 text-2xl"
          icon={faCakeCandles}
        />
      </div>
      <FontAwesomeIcon
        data-testid="sidebar-button"
        onClick={() => setIsOpen((isOpen) => !isOpen)}
        className="fixed top-4 right-4 text-3xl"
        icon={isOpen ? faClose : faBars}
      />
    </span>
  );
};
