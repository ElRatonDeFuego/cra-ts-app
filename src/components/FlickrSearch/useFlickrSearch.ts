import { CanceledError } from "axios";
import { useDebounce } from "hooks/useDebounce";
import { useEffectOnKeyPressed } from "hooks/useEffectOnKeyPressed";
import { useEffect, useMemo, useState } from "react";

// import/no-unused-modules requires non-baseUrl paths
import {
  buildFlickrPhotoURL,
  FlickrSearchURL,
  GenericErrorMessage,
  parseFlickrResponse,
} from "../../domain/Flickr";

import { useRequest } from "../../hooks/useRequest";

import type { FlickrPhoto } from "../../domain/Flickr";

export const useFlickrSearch = () => {
  const [photo, setPhoto] = useState<{ title?: string; URL?: string }>({});
  const [search, setSearch] = useState("");

  const debouncedSearch = useDebounce(search, 500);

  useEffectOnKeyPressed({ effect: () => setSearch(""), key: "Escape" });

  const url = `${FlickrSearchURL}${encodeURIComponent(debouncedSearch)}`;

  const { cancelRequest, data, error, loading, runRequest } =
    useRequest<string>({
      timeout: 5000,
      url,
    });

  useEffect(() => {
    if (debouncedSearch.length && search.length) {
      runRequest().catch(
        /* istanbul ignore next */ () => {
          // prevent unhandled exceptions
        }
      );
    }
  }, [debouncedSearch, runRequest, search]);

  const displayedResponse = useMemo(() => {
    if (error && !(error instanceof CanceledError)) {
      return error.message;
    }

    if (loading) {
      return "Loading..";
    }

    if (!(debouncedSearch.length && search.length && data)) {
      return "Please type in a search term above";
    }

    let result;

    try {
      result = parseFlickrResponse(data);
    } catch (parseError) {
      return GenericErrorMessage;
    }

    if (result.stat === "fail") {
      return GenericErrorMessage;
    }

    const { total } = result.photos;

    let responseMessage = `Found ${total} photo${total === 1 ? "" : "s"}`;

    if (result.photos.total > 0) {
      const randomPhoto = result.photos.photo[
        Math.floor(Math.random() * result.photos.photo.length)
      ] as FlickrPhoto;

      if (total !== 1) {
        responseMessage += " - here's one:";
      }

      setPhoto({
        title: randomPhoto.title,
        URL: buildFlickrPhotoURL(randomPhoto),
      });
    }

    return responseMessage;
  }, [data, debouncedSearch, error, loading, search]);

  return {
    cancelRequest, // used for tests
    displayedResponse,
    loading,
    photo,
    runRequest,
    search,
    setSearch,
  };
};
