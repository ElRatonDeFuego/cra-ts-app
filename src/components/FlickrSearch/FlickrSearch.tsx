import { faSearch, faSpinner } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import styles from "./FlickrSearch.module.scss";
import { Sidebar } from "./Sidebar";
import { useFlickrSearch } from "./useFlickrSearch";

export const FlickrSearch = () => {
  const { displayedResponse, loading, photo, runRequest, search, setSearch } =
    useFlickrSearch();

  return (
    <>
      <Sidebar />
      <span
        className={`items-center bg-gray-600 text-white flex flex-col
            justify-center min-h-screen max-w-full ${
              styles["flickr-search"] ?? /* istanbul ignore next */ ""
            }`}
      >
        <p className="font-light leading-none">Look for Flickr photos</p>
        <div className="bg-white text-gray-900 p-2 m-5 rounded-lg">
          <label htmlFor="search">
            <FontAwesomeIcon
              className={`ml-5 mr-5 text-xl ${loading ? "animate-spin" : ""}`}
              icon={loading ? faSpinner : faSearch}
            />
          </label>
          <input
            className="outline-none"
            data-testid="search-input"
            id="search"
            name="search"
            onChange={(event) => setSearch(event.target.value)}
            type="text"
            value={search}
          />
        </div>
        <p data-testid="displayed-response">{displayedResponse}</p>
        {photo.URL && (
          <>
            <img
              alt="first one found matching the search term"
              className="m-4 rounded-lg shadow-xl"
              data-testid="photo"
              src={photo.URL}
            />
            <p className="text-sm mt-4 px-1" data-testid="photo-title">
              {photo.title}
            </p>
            <button
              className="text-gray-900 px-5 py-3 m-5 cursor-pointer bg-white
              hover:bg-slate-100 focus:outline-none hover:shadow-lg
              hover:text-blue-800 active:bg-slate-300 active:animate-ping
              rounded-lg p-3 mt-6 shadow-lg uppercase tracking-wider
              font-medium"
              data-testid="refresh-button"
              onClick={() => {
                runRequest().catch(
                  /* istanbul ignore next */ () => {
                    // prevent unhandled exceptions
                  }
                );
              }}
            >
              Another one, please!
            </button>
          </>
        )}
      </span>
    </>
  );
};
