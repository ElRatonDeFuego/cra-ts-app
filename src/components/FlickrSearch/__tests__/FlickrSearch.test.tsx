import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import {
  buildFlickrPhotoURL,
  FlickrBaseURL,
  mockFlickrPhotos,
  mockFlickrResult,
} from "domain/Flickr";
import { rest } from "msw";
import { setupServer } from "msw/node";
import { FlickrSearch } from "../FlickrSearch";

import type { FlickrPhoto } from "domain/Flickr";

const server = setupServer();

// Establish API mocking before all tests.
beforeAll(() => server.listen());

beforeEach(() => {
  server.use(
    rest.get(FlickrBaseURL, (_req, res, ctx) =>
      res(ctx.delay(200), ctx.status(200), ctx.text(mockFlickrResult))
    )
  );
});

// Reset any request handlers that we may add during the tests,
// so they don't affect other tests.
afterEach(() => server.resetHandlers());

// Clean up after the tests are finished.
afterAll(() => server.close());

test("searches for photos with a debounce, clears search field on Escape", async () => {
  const view = render(<FlickrSearch />);

  const searchInput = screen.getByTestId("search-input");
  expect(searchInput).toBeInTheDocument();
  expect(searchInput.textContent).toEqual("");

  const displayedResponse = screen.getByTestId("displayed-response");
  expect(displayedResponse).toBeInTheDocument();

  const searchTerm = "house";

  userEvent.type(searchInput, searchTerm);

  expect(searchInput).toHaveValue(searchTerm);

  expect(displayedResponse).toHaveTextContent(
    "Please type in a search term above"
  );

  await waitFor(() => {
    expect(displayedResponse).toHaveTextContent("Loading..");
  });

  await waitFor(() => {
    expect(displayedResponse).toHaveTextContent(
      "Found 42 photos - here's one:"
    );
  });

  const photo = mockFlickrPhotos.photos.photo[0] as FlickrPhoto;

  const img: HTMLImageElement = screen.getByTestId("photo");
  expect(img).toBeInTheDocument();
  expect(img.src).toEqual(buildFlickrPhotoURL(photo));

  const title = screen.getByTestId("photo-title");
  expect(title).toBeInTheDocument();
  expect(title).toHaveTextContent(photo.title);

  const refreshButton = screen.getByTestId("refresh-button");
  expect(refreshButton).toBeInTheDocument();

  userEvent.click(refreshButton);

  await waitFor(() => {
    expect(displayedResponse).toHaveTextContent("Loading..");
  });

  await waitFor(() => {
    expect(displayedResponse).not.toHaveTextContent("Loading..");
  });

  userEvent.keyboard("{Escape}");

  expect(searchInput.textContent).toEqual("");

  expect(displayedResponse).toHaveTextContent(
    "Please type in a search term above"
  );

  const sidebarButton = screen.getByTestId("sidebar-button");
  expect(sidebarButton).toBeInTheDocument();

  userEvent.click(sidebarButton);

  expect(view).toMatchSnapshot();
});
