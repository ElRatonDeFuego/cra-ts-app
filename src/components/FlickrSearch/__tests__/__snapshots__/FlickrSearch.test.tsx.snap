// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searches for photos with a debounce, clears search field on Escape 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="text-emerald-50 opacity-95"
      >
        <div
          class="fixed top-0 right-0 p-6 pt-20 min-h-screen ease-in-out
                    duration-300 bg-slate-800 flex flex-col items-center
                    translate-x-0"
        >
          <h2>
            Useless animated sidebar!
          </h2>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-cake-candles text-center mt-8 text-2xl"
            data-icon="cake-candles"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M86.4 5.5L61.8 47.6C58 54.1 56 61.6 56 69.2V72c0 22.1 17.9 40 40 40s40-17.9 40-40V69.2c0-7.6-2-15-5.8-21.6L105.6 5.5C103.6 2.1 100 0 96 0s-7.6 2.1-9.6 5.5zm128 0L189.8 47.6c-3.8 6.5-5.8 14-5.8 21.6V72c0 22.1 17.9 40 40 40s40-17.9 40-40V69.2c0-7.6-2-15-5.8-21.6L233.6 5.5C231.6 2.1 228 0 224 0s-7.6 2.1-9.6 5.5zM317.8 47.6c-3.8 6.5-5.8 14-5.8 21.6V72c0 22.1 17.9 40 40 40s40-17.9 40-40V69.2c0-7.6-2-15-5.8-21.6L361.6 5.5C359.6 2.1 356 0 352 0s-7.6 2.1-9.6 5.5L317.8 47.6zM128 176c0-17.7-14.3-32-32-32s-32 14.3-32 32v48c-35.3 0-64 28.7-64 64v71c8.3 5.2 18.1 9 28.8 9c13.5 0 27.2-6.1 38.4-13.4c5.4-3.5 9.9-7.1 13-9.7c1.5-1.3 2.7-2.4 3.5-3.1c.4-.4 .7-.6 .8-.8l.1-.1 0 0 0 0s0 0 0 0s0 0 0 0c3.1-3.2 7.4-4.9 11.9-4.8s8.6 2.1 11.6 5.4l0 0 0 0 .1 .1c.1 .1 .4 .4 .7 .7c.7 .7 1.7 1.7 3.1 3c2.8 2.6 6.8 6.1 11.8 9.5c10.2 7.1 23 13.1 36.3 13.1s26.1-6 36.3-13.1c5-3.5 9-6.9 11.8-9.5c1.4-1.3 2.4-2.3 3.1-3c.3-.3 .6-.6 .7-.7l.1-.1c3-3.5 7.4-5.4 12-5.4s9 2 12 5.4l.1 .1c.1 .1 .4 .4 .7 .7c.7 .7 1.7 1.7 3.1 3c2.8 2.6 6.8 6.1 11.8 9.5c10.2 7.1 23 13.1 36.3 13.1s26.1-6 36.3-13.1c5-3.5 9-6.9 11.8-9.5c1.4-1.3 2.4-2.3 3.1-3c.3-.3 .6-.6 .7-.7l.1-.1c2.9-3.4 7.1-5.3 11.6-5.4s8.7 1.6 11.9 4.8l0 0 0 0 0 0 .1 .1c.2 .2 .4 .4 .8 .8c.8 .7 1.9 1.8 3.5 3.1c3.1 2.6 7.5 6.2 13 9.7c11.2 7.3 24.9 13.4 38.4 13.4c10.7 0 20.5-3.9 28.8-9V288c0-35.3-28.7-64-64-64V176c0-17.7-14.3-32-32-32s-32 14.3-32 32v48H256V176c0-17.7-14.3-32-32-32s-32 14.3-32 32v48H128V176zM448 394.6c-8.5 3.3-18.2 5.4-28.8 5.4c-22.5 0-42.4-9.9-55.8-18.6c-4.1-2.7-7.8-5.4-10.9-7.8c-2.8 2.4-6.1 5-9.8 7.5C329.8 390 310.6 400 288 400s-41.8-10-54.6-18.9c-3.5-2.4-6.7-4.9-9.4-7.2c-2.7 2.3-5.9 4.7-9.4 7.2C201.8 390 182.6 400 160 400s-41.8-10-54.6-18.9c-3.7-2.6-7-5.2-9.8-7.5c-3.1 2.4-6.8 5.1-10.9 7.8C71.2 390.1 51.3 400 28.8 400c-10.6 0-20.3-2.2-28.8-5.4V480c0 17.7 14.3 32 32 32H416c17.7 0 32-14.3 32-32V394.6z"
              fill="currentColor"
            />
          </svg>
        </div>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-xmark fixed top-4 right-4 text-3xl"
          data-icon="xmark"
          data-prefix="fas"
          data-testid="sidebar-button"
          focusable="false"
          role="img"
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M310.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L160 210.7 54.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L114.7 256 9.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 301.3 265.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L205.3 256 310.6 150.6z"
            fill="currentColor"
          />
        </svg>
      </span>
      <span
        class="items-center bg-gray-600 text-white flex flex-col
            justify-center min-h-screen max-w-full flickr-search"
      >
        <p
          class="font-light leading-none"
        >
          Look for Flickr photos
        </p>
        <input
          class="text-gray-900 p-2 m-5 rounded-lg outline-none"
          data-testid="search-input"
          name="search"
          type="text"
          value=""
        />
        <p
          data-testid="displayed-response"
        >
          Please type in a search term above
        </p>
        <img
          alt="first one found matching the search term"
          class="m-4 rounded-lg shadow-xl"
          data-testid="photo"
          src="http://farm66.staticflickr.com/65535/51263792541_a51806c4e6.jpg"
        />
        <p
          class="text-sm mt-4 px-1"
          data-testid="photo-title"
        >
          Photo data returned by msw
        </p>
        <button
          class="text-gray-900 px-5 py-3 m-5 cursor-pointer bg-white hover:bg-slate-100 focus:outline-none hover:shadow-lg hover:text-blue-800 active:bg-slate-300 active:animate-ping rounded-lg p-3 mt-6 shadow-lg uppercase tracking-wider font-medium"
          data-testid="refresh-button"
        >
          Another one, please!
        </button>
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="text-emerald-50 opacity-95"
    >
      <div
        class="fixed top-0 right-0 p-6 pt-20 min-h-screen ease-in-out
                    duration-300 bg-slate-800 flex flex-col items-center
                    translate-x-0"
      >
        <h2>
          Useless animated sidebar!
        </h2>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-cake-candles text-center mt-8 text-2xl"
          data-icon="cake-candles"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M86.4 5.5L61.8 47.6C58 54.1 56 61.6 56 69.2V72c0 22.1 17.9 40 40 40s40-17.9 40-40V69.2c0-7.6-2-15-5.8-21.6L105.6 5.5C103.6 2.1 100 0 96 0s-7.6 2.1-9.6 5.5zm128 0L189.8 47.6c-3.8 6.5-5.8 14-5.8 21.6V72c0 22.1 17.9 40 40 40s40-17.9 40-40V69.2c0-7.6-2-15-5.8-21.6L233.6 5.5C231.6 2.1 228 0 224 0s-7.6 2.1-9.6 5.5zM317.8 47.6c-3.8 6.5-5.8 14-5.8 21.6V72c0 22.1 17.9 40 40 40s40-17.9 40-40V69.2c0-7.6-2-15-5.8-21.6L361.6 5.5C359.6 2.1 356 0 352 0s-7.6 2.1-9.6 5.5L317.8 47.6zM128 176c0-17.7-14.3-32-32-32s-32 14.3-32 32v48c-35.3 0-64 28.7-64 64v71c8.3 5.2 18.1 9 28.8 9c13.5 0 27.2-6.1 38.4-13.4c5.4-3.5 9.9-7.1 13-9.7c1.5-1.3 2.7-2.4 3.5-3.1c.4-.4 .7-.6 .8-.8l.1-.1 0 0 0 0s0 0 0 0s0 0 0 0c3.1-3.2 7.4-4.9 11.9-4.8s8.6 2.1 11.6 5.4l0 0 0 0 .1 .1c.1 .1 .4 .4 .7 .7c.7 .7 1.7 1.7 3.1 3c2.8 2.6 6.8 6.1 11.8 9.5c10.2 7.1 23 13.1 36.3 13.1s26.1-6 36.3-13.1c5-3.5 9-6.9 11.8-9.5c1.4-1.3 2.4-2.3 3.1-3c.3-.3 .6-.6 .7-.7l.1-.1c3-3.5 7.4-5.4 12-5.4s9 2 12 5.4l.1 .1c.1 .1 .4 .4 .7 .7c.7 .7 1.7 1.7 3.1 3c2.8 2.6 6.8 6.1 11.8 9.5c10.2 7.1 23 13.1 36.3 13.1s26.1-6 36.3-13.1c5-3.5 9-6.9 11.8-9.5c1.4-1.3 2.4-2.3 3.1-3c.3-.3 .6-.6 .7-.7l.1-.1c2.9-3.4 7.1-5.3 11.6-5.4s8.7 1.6 11.9 4.8l0 0 0 0 0 0 .1 .1c.2 .2 .4 .4 .8 .8c.8 .7 1.9 1.8 3.5 3.1c3.1 2.6 7.5 6.2 13 9.7c11.2 7.3 24.9 13.4 38.4 13.4c10.7 0 20.5-3.9 28.8-9V288c0-35.3-28.7-64-64-64V176c0-17.7-14.3-32-32-32s-32 14.3-32 32v48H256V176c0-17.7-14.3-32-32-32s-32 14.3-32 32v48H128V176zM448 394.6c-8.5 3.3-18.2 5.4-28.8 5.4c-22.5 0-42.4-9.9-55.8-18.6c-4.1-2.7-7.8-5.4-10.9-7.8c-2.8 2.4-6.1 5-9.8 7.5C329.8 390 310.6 400 288 400s-41.8-10-54.6-18.9c-3.5-2.4-6.7-4.9-9.4-7.2c-2.7 2.3-5.9 4.7-9.4 7.2C201.8 390 182.6 400 160 400s-41.8-10-54.6-18.9c-3.7-2.6-7-5.2-9.8-7.5c-3.1 2.4-6.8 5.1-10.9 7.8C71.2 390.1 51.3 400 28.8 400c-10.6 0-20.3-2.2-28.8-5.4V480c0 17.7 14.3 32 32 32H416c17.7 0 32-14.3 32-32V394.6z"
            fill="currentColor"
          />
        </svg>
      </div>
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-xmark fixed top-4 right-4 text-3xl"
        data-icon="xmark"
        data-prefix="fas"
        data-testid="sidebar-button"
        focusable="false"
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M310.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L160 210.7 54.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L114.7 256 9.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 301.3 265.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L205.3 256 310.6 150.6z"
          fill="currentColor"
        />
      </svg>
    </span>
    <span
      class="items-center bg-gray-600 text-white flex flex-col
            justify-center min-h-screen max-w-full flickr-search"
    >
      <p
        class="font-light leading-none"
      >
        Look for Flickr photos
      </p>
      <input
        class="text-gray-900 p-2 m-5 rounded-lg outline-none"
        data-testid="search-input"
        name="search"
        type="text"
        value=""
      />
      <p
        data-testid="displayed-response"
      >
        Please type in a search term above
      </p>
      <img
        alt="first one found matching the search term"
        class="m-4 rounded-lg shadow-xl"
        data-testid="photo"
        src="http://farm66.staticflickr.com/65535/51263792541_a51806c4e6.jpg"
      />
      <p
        class="text-sm mt-4 px-1"
        data-testid="photo-title"
      >
        Photo data returned by msw
      </p>
      <button
        class="text-gray-900 px-5 py-3 m-5 cursor-pointer bg-white hover:bg-slate-100 focus:outline-none hover:shadow-lg hover:text-blue-800 active:bg-slate-300 active:animate-ping rounded-lg p-3 mt-6 shadow-lg uppercase tracking-wider font-medium"
        data-testid="refresh-button"
      >
        Another one, please!
      </button>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
